src/Application/Contracts/IUserService.cs

```diff
@@ -8,7 +8,7 @@ public interface IUserService
     {
         Task<Response<string>> LoginAsync(string email, string password);
         Task<Response<NhsLoginResponse>> NhsLoginAsync(string code, string redirectUrl);
-        Task<Response<SignUpResponse>> NhsSignUpAsync(bool consent, string token);
+        Task<Response<SignUpResponse>> NhsSignUpAsync(bool consent, string selectedLanguage, string token);
```

Comment:
I would be inclined to added the new parameter at the end of the signature to make maintenance easier (unless there is a good reason to add it sooner, like importance). Especially when the types are the same, easy to get `token` and `selectedLanguage` mixed up when updating existing call sites.

I would recommend using the `CultureInfo selectedLocale` instead.

Reply:
I just grouped the first 2 together as these are both "user actions".  I have updated the type to CultureInfo and now the work is done I think the risk is minimised, changing back would be unnecessary but I will keep in mind for future dev